DDD-13.md

```diff
@@ -0,0 +1,595 @@
+
+* name
+* description
+* type
+* default value
+* validator
+
+These properties are generally documented in the component docs, which is sufficient if you configure it in the standard way (Kafka Connect, Debezium Server, Debezium engine), but this is not the case for the Debezium Platform.
+
+The UI of the Debezium Platform (stage) needs a way to get all the possible configuration properties declared by a component to easily allow users to configure them.
+
+This document will present how this can be achieved.
+
+## Goals
+
+The goal is to define a descriptor for Connectors, Transformations, and Sinks (and storage?) and how this will be created and exposed to the stage.
+
+## Descriptors format
+
+Descriptor format should be expressive and human writable/readable. The first thought goes to JSON and YAML, but also a custom DSL can be an option.
+Let's explore how the different formats appear for describing connector properties.
+
+```json
+{
+  "name": "PostgreSQL Connector",
+  "type": "source-connector",
+  "version": "1.0.0",
+  "metadata": {
+    "description": "Captures changes from a PostgreSQL database",
+    "tags": ["database", "postgresql", "cdc"]
+  },
+  "properties": [
+    {
+      "name": "topic.prefix",
+      "type": "string",
+      "required": true,
+      "display": {
+        "label": "Topic prefix",
+        "description": "Topic prefix that identifies and provides a namespace for the particular database server/cluster that is capturing changes. The topic prefix should be unique across all other connectors, since it is used as a prefix for all Kafka topic names that receive events emitted by this connector. Only alphanumeric characters, hyphens, dots and underscores must be accepted.",
+        "group": "Connection",
+        "groupOrder": 0,
+        "width": "medium",
+        "importance": "high"
+      },
+      "validation": [
+        {
+          "type": "regex",
+          "pattern": "^[a-zA-Z0-9._-]+$",
+          "message": "Only alphanumeric characters, hyphens, dots and underscores are allowed"
+        }
+      ]
+    },
+    {
+      "name": "table.include.list",
+      "type": "list",
+      "display": {
+        "label": "Include Tables",
+        "description": "The tables for which changes are to be captured",
+        "group": "Filters",
+        "groupOrder": 2,
+        "width": "long",
+        "importance": "high"
+      },
+      "validation": [
+        {
+          "type": "regex-list",
+          "message": "Each item must be a valid regular expression"
+        }
+      ]
+    },
+     {
+        "name": "connection.adapter",
+        "type": "string",
+        "display": {
+           "label": "Connector adapter",
+           "description": "The adapter to use when capturing changes from the database.",
+           "group": "Connection advanced",
+           "groupOrder": 1,
+           "width": "medium",
+           "importance": "high"
+        },
+        "valueDependants": [
```

Comment:
While creating this example seems that the concept of `dependant` has more sense only related to a `value` and not to a field itself. 

If I'm correct, in the current implementation the dependents declared with `public Field withDependents(String... dependents)` will be used in the `Field.validate(...)` to validate all dependents fields. 

So suppose I have a property `prop1` with `prop2` as dependent, validating the `prop1` will trigger the validation of `prop2`. Should this happen in any case?

So to me seems to have not so much sense.

Reply:
So do you propose to keep validation only based on the value of the field but not on the field itself?

Reply:
No, what I'm trying to say is that the `withDependents(String... dependents)` seems not to be so useful while instead `io.debezium.config.Field.withDependents(String fieldValue, String... dependents)` this is the one we really want. 

I don't see any useful usage, as of now, of the `withDependents(String... dependents)`. Maybe I'm missing something.